### ISSUE TITLE
Asynchronos PDR-ACK

### ISSUE THREAD

**Issue by inesrob** [roll-wg/dao-projection#12]
Source:  https://mailarchive.ietf.org/arch/msg/roll/z5-8IIxs9sWQueSJJ1ksHssNlRM/

Dear all: we had this thread as part of Li's review:

> [Li] So PDR-ACK is for original PDR. And for maintain or tear down case, Root should use No-Path P-DAO to terminate P-Route. Correct?

[PT] yes: the NP DAO cleans the state in the network. It removes the Track. The PDR ACK signals to the Ingress that it is being done and why.  Maybe we need to explain that those 2 may be asynchronous to one another. I agree with you that we can probably think this through a little more.

  *   E.g., should the PDR ACK trigger a DCO (RFC 9009) ?
  *   Could the PDR ack signal status that is not the death of the Track but other stuff?

Basically we need to clarify when the main Root sends an async PDR ACK, what status go on there and what the reaction is by the Track Ingress based on the status (including unknown).

Comments welcome!

Pascal

From: Roll <roll-bounces@ietf.org> On Behalf Of Pascal Thubert (pthubert)
Sent: vendredi 14 janvier 2022 9:18
To: Routing Over Low power and Lossy networks <roll@ietf.org>
Subject: Re: [Roll] review of dao-projection -22

Hello Li:

> [Li] Yes. It's more clear. So PDR-ACK is for original PDR. And for maintain or tear down case, Root should use No-Path P-DAO to terminate P-Route. Correct?

[PT] yes: the NP DAO cleans the state in the network. It removes the Track. The PDR ACK signals to the Ingress that it is being done and why.  Maybe we need to explain that those 2 may be asynchronous to one another. I agree with you that we can probably think this through a little more. E.g., should the PDR ACK trigger a DCO (RFC 9009) ? Could the PDR ack signal status that is not the death of the Track but other stuff?

> [Li] Maybe it depends on implement of PCE?  I'd like to keep the flexibility because there is a user case for street lighting. Some controllers(Ingress nodes) need to light the street lighting linearly with low-latency.
[PT] sure, at the expense of complexity in signaling and processing in the nodes. We need to  be specific on the scenarios and status codes. The PDR forms a single Track since the TrackID is provided. Since signaling is involved we need a story for all those questions:

  *   What if there's a path to only a subset? Should the root form more than one track?
  *   Or deny with your special status but then what can the source do after that? Try all combinations?
  *   What is the best path to one target cannot reach all targets but a lower quality path can? Should we use that one?
  *   Maybe there could be one main target and additional desirable targets? In that case the Root could form the track and indicate which of the optional targets are effectively reachable via the track to the main target.

> [Li] Some physical layer measurements such as RSSI/LQI have forward and reverse direction. Can it indicate symmetrical?
                  In layer 3, receiving DIO/NS messages from each other indicate symmetrical
There's a nuance between bidir and symmetrical.  Maybe the ping works but the link quality is very different in both directions. If so the Rank increment computation would differ widely and we would need both sides.

We need separate threads to solve these issues. Let me start them

Keep safe!

Pascal

Pascal

**Comment by pthubert** [comment]
[From:]() Roll <roll-bounces@ietf.org> On Behalf Of Pascal Thubert (pthubert)
Sent: jeudi 24 février 2022 14:36
To: ROLL WG (roll@ietf.org) <roll@ietf.org>
Subject: Re: [Roll] Asynchronous PDR-ACK procedure (Was: review of dao-projection -22)

Dear all

This is now issue 12 in github https://github.com/roll-wg/dao-projection/issues/12

I see the power of an async PDR ack but have no use case for it. The question was sent to the list a month ago and there was no suggestion; I propose that we hold the odea for the day we need it and keep things ass is for now.

Objection?

Keep safe;

Pascal


From: Pascal Thubert (pthubert) 
Sent: vendredi 14 janvier 2022 10:15
To: ROLL WG ([roll@ietf.org](mailto:roll@ietf.org)) <[roll@ietf.org](mailto:roll@ietf.org)>
Cc: Li Zhao (liz3) <[liz3@cisco.com](mailto:liz3@cisco.com)>
Subject: Asynchronous PDR-ACK procedure (Was: [Roll] review of dao-projection -22)

Dear all: we had this thread as part of Li’s review:

> [Li] So PDR-ACK is for original PDR. And for maintain or tear down case, Root should use No-Path P-DAO to terminate P-Route. Correct?

[PT] yes: the NP DAO cleans the state in the network. It removes the Track. The PDR ACK signals to the Ingress that it is being done and why.  Maybe we need to explain that those 2 may be asynchronous to one another. I agree with you that we can probably think this through a little more. 
•	E.g., should the PDR ACK trigger a DCO (RFC 9009) ? 
•	Could the PDR ack signal status that is not the death of the Track but other stuff?


Basically we need to clarify when the main Root sends an async PDR ACK, what status go on there and what the reaction is by the Track Ingress based on the status (including unknown).

Comments welcome!

Pascal
